src/org/authInfo.ts

```diff
@@ -574,6 +512,14 @@ export class AuthInfo extends AsyncOptionalCreatable<AuthInfo.Options> {
     return opts;
   }
 
+  public getClientId(): string {
+    return this.getFields()?.clientId || DEFAULT_CONNECTED_APP_INFO.legacyClientId;
```

Comment:
@mdonnalley Why are defaulting to legacyClientId. This implies that if customers have used the current default clientId then `this.getFields()?.clientId` will always be truthy.

Reply:
I think the idea is that if you're auth'ing for the first time then `getFields().clientId` will be falsy, in which case we default to the legacy client id